src/significance.jl

```diff
@@ -0,0 +1,107 @@
+"""
+
+    measure_significance(res, significance_metric)
+
+Compute the `significance_metric` on the results `res` of an indicator evolution
+analysis.
+"""
+@inline function measure_significance(
+    res::IndicatorEvolutionResults{T},
+    significance_metric::Function,
+) where {T<:AbstractFloat}
+    return measure_significance(
+        res.x_evolution,
+        res.surr_x_evolution,
+        significance_metric,
+    )
+end
+
+@inline function measure_significance(
+    x_evolution::AbstractVector{T},
+    surr_x_evolution::AbstractMatrix{T},
+    significance_metric::Function,
+) where {T<:AbstractFloat}
+    significance = similar(x_evolution)
+    for j in eachindex(significance)
+        significance[j] = significance_metric(
+            x_evolution[j],
+            view(surr_x_evolution, :, j),
+        )
+    end
+    return significance
+end
+
+#####################################################
+# Percentile metrics
+#####################################################
+"""
+
+    percentile_idx(s; p, symmetric)
+
+If `symmetric == false`, return indices of `0`-th and `p*100`-th percentile of `s`.
+If `symmetric == true` and `p=0.95`, return indices of `2.5`-th and `97.5`-th percentile.
+"""
+@inline function percentile_idx(
+    s::AbstractVector{T};
+    p::T = T(0.95),
```

Comment:
Why should the percentile match type with `x`? I think it shouldn't. In fact, there is no real reason to specify the type of `p` here, and just return `T(i_lo), T(i_hi)` at the end.

Reply:
corrected. also `i_lo` and `i_hi` are integers